[ReganChoy] iP

ReganChoy opened the pull request:
# DukeXAerial

>Time is of the Essence

- Minimal Crashes
* FAST

1. Early Access 🧣 
2. ~~Lifetime Support~~

Some application commands include:
```
todo <Task>
deadline <Task> dd/mm/yyyy hh:mm
bye
list
mark <Task Number>
```

- [x] Early build
- [ ] Full Test

>[!CAUTION]
>DO NOT AMEND THE TXT FILE TO PREVENT DATA CORRUPTION

[Main Project Source](https://nus-cs2103-ay2324s2.github.io/website/admin/index.html)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

ReganChoy committed:
Added Name, Greeting and Bye

ReganChoy committed:
Added Echo

ReganChoy committed:
Added: List and Add methods

ReganChoy committed:
Added: Task class, Mark and Unmark methods

ReganChoy committed:
Added: Subclassses of Task (ToDo, Event, Deadline)

ReganChoy committed:
Edited: input.txt and Expected.txt

ReganChoy committed:
Added: Exception Handling for Methods

ReganChoy committed:
Added: Delete method

ReganChoy committed:
Implemented Load and Save functions
Re-organized code into different classes from Duke class

ReganChoy committed:
Converted DateTime format from String to DateTme class

ReganChoy committed:
Increased OOP by creating Parser, TaskList, Ui and Storage classes

ReganChoy committed:
Packaged all Classes into a single package named duke

ReganChoy committed:
Update duke

ReganChoy committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ReganChoy committed:
Set up and run Duke with gradle
Edit build.gradle file to run

ReganChoy committed:
Add junit test cases for ToDo and Event

ReganChoy committed:
Edit build.gradle file for jar

ReganChoy committed:
Edit build.gradle file to generate only duke.jar file

ReganChoy committed:
Edit all files to be inline with Java Coding Standards

ReganChoy committed:
Add JavaDocs for all public classes and methods

ReganChoy committed:
Add find method to find tasks with matching description

ReganChoy committed:
Merge branch 'branch-Level-9'

ReganChoy committed:
Merge branch 'branch-A-JavaDoc' with master

ReganChoy committed:
Merge branch 'branch-A-CodingStandard with master

ReganChoy committed:
Fix Java Standards for TaskList findList method

ReganChoy committed:
Add GUI and increase class abstraction

ReganChoy committed:
Add checkstyle files

ReganChoy committed:
Fix checkstyle errors

solomonng2001 commented on docs/README.md:
```diff
@@ -1,29 +1,30 @@
-# User Guide
+# Duke User Guide
```
why not call it Gundam? 😜

solomonng2001 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,63 @@
+     * Informs the user a new Deadline has been created.
+     * @return String informing a new Deadline has been created.
+     */
+    public String announcement() {
```
Can this be renamed to an action? getAnnouncment etc? I noticed the announcement() method in other files as well

ReganChoy replied:
Good idea!

solomonng2001 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,63 @@
+     * Returns a string representation of the Deadline for txt file saving purposes.
+     * @return String representation of the Deadline for txt file saving purposes.
+     */
+    public String saveString() {
```
Is there an easier way other than using multiple concatenation operators? Would the following work for you? Actually, I think this is personally preference, but would like to let you know of alternatives. Concatenation could be more readable too, since variables and string can be read sequentially.

<img width="465" alt="Screenshot 2024-02-07 at 7 07 14 PM" src="https://github.com/nus-cs2103-AY2324S2/ip/assets/66112309/6a786c30-1f83-46e4-8922-c79a991de553">

I noticed similar instances elsewhere.

solomonng2001 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,102 @@
+        this.command = "";
+        this.secondaryInput = "";
+
+        if (commandInput.equals("save")) {
```
Is there a way to keep track of loose strings like "save" in a constant or enum? A central area to store such strings as constants can make a large codebase more readable. The benefits are as follows
1. can go to a command enum file or command constants to see all types of commands
2. can change all instances of the same "save" command, by changing the string saved in a constant or enum to "update"

solomonng2001 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,110 @@
+     * Method generates the string of the task in format for file txt saving.
+     * @return String representation of the task in the format for file txt saving.
+     */
+    public String saveString() {
```
Should this be saved as a constant (static final) instead, since it does not operate on the string but only returns it as it is?

Noticed similar instances elsewhere, eg with announcment()

solomonng2001 reviewed (commented):
Good job Regan, just some nits.

Strong OOP and good handling of exceptions. I focused more on syntax and standards, rather than functional. Hope they are useful. Additionally, some of the methods could be made private without changing functionality of the code, but up to you if you would like to implement it that way.

Overall great job! 😄

jimseah-0116 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,102 @@
+     * @param duke
+     * @param taskList
+     */
+    public String input(String commandInput, Duke duke, TaskList taskList) {
```
Since this is a method, following the coding convention, maybe you could give the method a verb name instead?
For example, public String parseInput(...) could work.

jimseah-0116 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,83 @@
+
+        Scanner scanner = new Scanner(file);
+        while (scanner.hasNext()) {
+            String inputLine = scanner.nextLine();
+            String[] inputComponents = inputLine.split("\\|", 5);
```
Variable names are in camelCase. Great!

jimseah-0116 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,76 @@
+
+    /**
+     * Constructor for Event Class.
+     * @param task
+     * @param taskType
+     * @param start
+     * @param end
```
Perhaps these @param tags may be unnecessary?

jimseah-0116 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,166 @@
+
+    /**
+     * Adds new task to the TaskListt.
+     * @param first
+     * @param second
```
The parameter names are a bit confusing - perhaps you may want to specify that 'first' is the Task type and second is the rest of the String?

ReganChoy replied:
Good suggestion!

jimseah-0116 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,58 @@
+    private Parser parser = new Parser();
+    private Scanner scanner;
+    private boolean isEnded = false;
+    private final String PATH = "data/tasks.txt";
```
Constant PATH is in all upper case letters. Great!

jimseah-0116 commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+ * This control represents a dialog box consisting of an ImageView to represent the speaker's face and a label
+ * containing text from the speaker.
+ */
+public class DialogBox extends HBox {
```
Class name is PascalCase. Great!

ReganChoy committed:
Abstract checkInput function in Parser class into different check methods

Each command now has its own checker to verify the command format.
This makes it easier to add and test new commands.

ReganChoy committed:
Merge pull request #1 from ReganChoy/branch-A-CodeQuality

Abstract checkInput function in Parser class

ReganChoy committed:
Add assertions into Parser class

Assertions check if deadline's and event's Date and Time are preceded with /

ReganChoy committed:
Merge pull request #2 from ReganChoy/branch-A-Assertions

Add assertions into Parser class

ReganChoy committed:
Add postpone functionality

Users can now postpone deadlines and events.
ToDo tasks cannot be postpone as they are not time sensitive.

ReganChoy committed:
Fix checkstyle and function naming issues

ReganChoy committed:
Fix JavaDocs method header and add Ui.png

ReganChoy committed:
Add draft for UserGuide

ReganChoy committed:
Build user guide

ReganChoy committed:
Update user guide formatting

ReganChoy committed:
Fix user guide formatting

ReganChoy committed:
Fix user guide formatting

ReganChoy committed:
Fix formatting issues for user guide

ReganChoy committed:
Fix incorrect instructions for time format

ReganChoy committed:
Add more J-Unit test cases for Parser Class

ReganChoy committed:
Fix user guide formatting issues

ReganChoy committed:
Fix checkstyle issues

ReganChoy committed:
Update jar file and include setting up steps in user guide

ReganChoy committed:
Format user guide
